### PR TITLE
Gam 13

### PR DESCRIPTION
**Pull request by Christian-Rades**
Rewrote matchController and implemented a new "new match"-form

### CODE REVIEW

File: app/Resources/views/modals/newMatch.html.twig [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+{% block modalTitle %}
+    Enter match results
+{% endblock %}
+
+{% block modalbegin %}
+    {{ form_start(form) }}
+{% endblock %}
+
+{% block modal_content %}
+    {#{{ form_row(form.scores) }}#}
```
Comment (tobiasberge) [inline app/Resources/views/modals/newMatch.html.twig]: Please remove this

File: web/assets/js/dynamicScoreList.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    // increase the index with one for the next item
+    $collectionHolder.data('index', index + 1);
+
+    // Display the form in the page in an div, before the "Add a tag" link div
+    var $newFormSpan = $('<div class="row form-group"></div>').append(newForm);
+    $newFormSpan.append('<div class="col-md-2"><a href="#" class="btn btn-block btn-danger btn-md remove-tag"><span class="glyphicon glyphicon-remove" aria-hidden="true"></span></a></div>');
+    $newLinkDiv.before($newFormSpan);
+    if ($collectionHolder.find(':input').length > 0){
+        $saveButton.removeAttr('disabled');
+        console.log("derp");
```
Comment (tobiasberge) [inline web/assets/js/dynamicScoreList.js]: Please remove the derp 🤣

File: web/assets/js/dynamicScoreList.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+
+    $('.remove-tag').click(function(e) {
+        e.preventDefault();
+        $(this).parent().parent().remove();
+        if ($collectionHolder.find(':input').length == 0){
+            $saveButton.attr('disabled',true);
+            console.log("derp");
```
Comment (tobiasberge) [inline web/assets/js/dynamicScoreList.js]: 😄 🤣

File: src/AppBundle/MatchLog.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+    /**
+     * @param Player $player
+     * @return Match[]
+     */
+    public function getMatchesByPlayer(Player $player){
+        $scores = $this->scoreRepo->findBy($player);
+        $matches = [];
+
+        foreach ($scores as $score)
```
Comment (dbeier) [inline src/AppBundle/MatchLog.php]: foreach () {

File: src/AppBundle/MatchLog.php [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        $this->matchRepo = $matchRepository;
+        $this->gameRepo = $gameRepository;
+        $this->scoreRepo = $scoreRepository;
+    }
+
+    /**
+     * @param Player $player
+     * @return Match[]
+     */
+    public function getMatchesByPlayer(Player $player){
```
Comment (dbeier) [inline src/AppBundle/MatchLog.php]: space beyond ) and { ){ - ) {

**Pull request merged**